WordPress/gutenberg
# Docs: Add missing data module references

## What?
This PR adds a new reference to [the Data Module Reference documentation](https://github.com/WordPress/gutenberg/tree/trunk/docs/reference-guides/data).

## Why?

I noticed that the following packages with store are not explicitly mentioned in the documentation.

- block-directory
- customize-widgets
- ~~edit-navigation~~ (excluded due to private package)
- edit-site
- edit-widgets
- ~~interface~~ (excluded due to internal package)
- preferences
- reusable-blocks
- rich-text

These packages are not documented, so developers need to check the source code and README directly in the GitHub repository.

## How?

With the exception of the `package` (which was clearly marked as experimental), I created markdown file templates and generated documentation by `npm run api-docs:ref` command.

At the same time, I updated the related TOC and JSON data.

## Testing Instructions

Check [the New documentation page](https://github.com/WordPress/gutenberg/blob/docs/add-references/docs/reference-guides/data/README.md).
